### DISCUSSION TITLE
Different translation method for `text-editor` addon?

### DISCUSSION THREAD

**Discussion by nexensys** [ScratchAddons/ScratchAddons#2860] [Ideas]
So I am working on making the `text-editor` addon, mainly the parsing methods right now. I realized that translations would be pretty hard to do with the `msg` api, and so I started looking through the [scratchblocks translation files](https://github.com/scratchblocks/scratchblocks/tree/master/locales). I would like to use these to lessen the workload, and I can do something like:

```js
export default async function ({ msg }) {
  ...
  const syntax = await import(`./locales/${msg.locale}`);
  ...
}
```

then do some work to prepare it for parsing. I could make it use the `msg` api, but it would take a lot more work than just downloading some pre-made translations and tweaking them. What I really want to know is this: Will this cause any issues with the current code and/or would this be better if I used the `msg` api?

**Comment by apple502j** [comment]
It depends on what you need, but `addon.tab.scratchMessage` perhaps? You probably don't need to include scratchblocks translation file.

**Reply by nexensys** [reply]
Does it have it for all of the blocks? I need the translations for pretty much every part of the BBcode, including the @labels and aliases...
